applications/asset_tracker_v2/src/modules/cloud_module.c

```diff
@@ -435,6 +526,27 @@ static void connect_cloud(void)
 	k_work_reschedule(&connect_check_work, K_SECONDS(backoff_sec));
 }
 
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+void pgps_handler(enum nrf_cloud_pgps_event event, struct nrf_cloud_pgps_prediction *prediction)
+{
+	int err;
+
+	if (event != PGPS_EVT_AVAILABLE) {
+		return;
+	}
+
+	err = nrf_cloud_pgps_inject(prediction, &agps_request, NULL);
```

Comment:
In Asset Tracker v1, I did these two steps in a work handler started from this pgps_handler, because the stack usage can sometimes be large, and the GPS driver thread stack isn't always large enough.  You might get a crash here like this.

Reply:
In the case of v2 all this functionality is called from a separate thread in the cloud module.